src/lib/get-subdomain-parts.ts

```diff
@@ -7,10 +7,11 @@ export interface UrlParts {
 }
 
 export function getSubdomainParts (urlString: string): UrlParts {
-  const labels = new URL(urlString).hostname.split('.')
+  const url = new URL(urlString)
+  const labels = url.host.split('.')
   let id: string | null = null
   let protocol: string | null = null
-  let parentDomain: string = urlString
+  let parentDomain: string = url.host
```

Comment:
fixes an issue where parentDomain is not set correctly if its not a legitimate subdomain. there are tests added for this.